vms/platformvm/txs/mempool/mempool.go

```diff
@@ -5,58 +5,109 @@ package mempool
 
 import (
 	"errors"
+	"fmt"
+	"sync"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/heap"
+	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/ava-labs/avalanchego/vms/platformvm/config"
 	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs/fee"
 
 	txmempool "github.com/ava-labs/avalanchego/vms/txs/mempool"
 )
 
 var (
-	_ Mempool = (*mempool)(nil)
-
 	ErrCantIssueAdvanceTimeTx     = errors.New("can not issue an advance time tx")
 	ErrCantIssueRewardValidatorTx = errors.New("can not issue a reward validator tx")
+
+	_ meterer = (*noMeter)(nil)
 )
 
-type Mempool interface {
-	txmempool.Mempool[*txs.Tx]
+type Tx struct {
+	*txs.Tx
+	Complexity gas.Dimensions
+	Gas        gas.Gas
+}
+
+type meterer interface {
+	Meter(tx txs.UnsignedTx) (gas.Dimensions, gas.Gas, error)
+}
+
+type noMeter struct{}
+
+func (noMeter) Meter(txs.UnsignedTx) (gas.Dimensions, gas.Gas, error) {
+	return gas.Dimensions{}, 0, nil
+}
 
-	// RequestBuildBlock notifies the consensus engine that a block should be
-	// built. If [emptyBlockPermitted] is true, the notification will be sent
-	// regardless of whether there are no transactions in the mempool. If not,
-	// a notification will only be sent if there is at least one transaction in
-	// the mempool.
-	RequestBuildBlock(emptyBlockPermitted bool)
+type dynamicMeter struct {
+	weights gas.Dimensions
 }
 
-type mempool struct {
-	txmempool.Mempool[*txs.Tx]
+func (d dynamicMeter) Meter(tx txs.UnsignedTx) (gas.Dimensions, gas.Gas, error) {
+	c, err := fee.TxComplexity(tx)
+	if err != nil {
+		return gas.Dimensions{}, 0, err
+	}
 
+	g, err := c.ToGas(d.weights)
+	if err != nil {
+		return gas.Dimensions{}, 0, err
+	}
+
+	return c, g, nil
+}
+
+type Mempool struct {
+	mempool  txmempool.Mempool[*txs.Tx]
```

Comment:
I'm surprised we still use the other mempool implementation. Is there a reason why we want to depend on the FIFO mempool implementation here?

Reply:
This was to keep the current behavior of the mempool (e.g caching drop reasons, checking for conflicts) and just add ordering to make the PR easier to review. I'm also open to implementing this as a separate type.

Reply:
Removed dependency on common mempool